File: example/legacy_not_compliant/fubon_app_ts/workflows/submit_fubon_insurance/flow.ts
import { Context } from 'platform_sdk_ts';
import * as path from 'path';

// 假設這裡有引入 lib 裡面的 domain logic (原 repo 裡有許多這類的工具)
// import { fetchNamelist } from '../../lib/dap-source';
// import { applyInsurance } from '../../lib/playwright/fubon';

export async function flow(ctx: Context, inputs: any = {}) {
  const company = ctx.secrets.get("FUBON_COMPANY");
  const user = ctx.secrets.get("FUBON_USER");
  const pass = ctx.secrets.get("FUBON_PASSWORD");
  const driveFolderId = ctx.secrets.get("FUBON_NAMELIST_DRIVE_FOLDER_ID");

  ctx.logger.info("=========================================");
  ctx.logger.info("🚀 啟動富邦自動投保流程 (TS Version)");
  
  if (!company || !user || !pass) {
    throw new Error("Missing Fubon credentials in secrets!");
  }

  try {
    // 1. 取得名單 (模擬)
    ctx.logger.info(`從 Google Drive [${driveFolderId}] 下載今日名單...`);
    // await fetchNamelist(driveFolderId);
    
    // 2. 登入與填表 (這裡示範如何無縫銜接原有的 playwright page 操作)
    ctx.logger.info("啟動無頭瀏覽器進行投保...");
    const page = await ctx.browser!.newPage();
    
    await page.goto("https://b2b.518fb.com/login");
    // await page.fill('#username', user);
    // await applyInsurance(page, namelist);
    
    ctx.logger.info("✅ 投保成功，準備發送 Slack 通知...");
    ctx.notify.slack("✅ [富邦旅平險] 今日投保已全數完成。");
    
    return { status: "success" };
    
  } catch (error: any) {
    ctx.logger.error(`投保失敗: ${error.message}`);
    ctx.notify.slack(`❌ [富邦旅平險] 投保失敗: ${error.message}`);
    throw error;
  }
}

File: example/legacy_not_compliant/ticket_bot/workflows/process_be2_ticket/flow.py
"""
BE2 工單處理 - 核心流程
瀏覽器一律經 ctx.browser(受管 Playwright headless);登入/建單細節待從原 repo 移植。
"""


def flow(ctx, order_id, wantan_type="mansatisfied", follow_type="page"):
    # 從 ctx 取得憑證,不直接讀 os.environ
    username = ctx.secrets.get("vault://ticket-bot/BE2_USERNAME")
    password = ctx.secrets.get("vault://ticket-bot/BE2_PASSWORD")

    page = ctx.browser.new_page()
    try:
        with ctx.log.step("login"):
            ctx.log.info(f"開始處理訂單: {order_id}")
            page.goto("https://be2.kkday.com/login")
            # TODO: 從原 kkday-ticket-bot 移植登入邏輯
            # page.fill("input[type='email']", username)
            # page.fill("input[type='password']", password)

        with ctx.log.step("create_ticket"):
            # TODO: 移植找訂單、建工單、寫後拋邏輯
            ticket_id = f"TICK-{order_id[-4:]}"
            ctx.log.info(f"工單建立成功: {ticket_id}", ticket_id=ticket_id)

        ctx.notify(f"✅ 訂單 {order_id} 工單完成: {ticket_id}")
        return {"status": "success", "ticket_id": ticket_id}
    finally:
        # 失敗時不在這裡吞錯 — 拋出讓框架做失敗三件套(截圖/告警/狀態頁)
        ctx.browser.close()
